### PR TITLE
gcoap: enhance gcoap_req_init

### PR DESCRIPTION
**Pull request by smlng**
### Contribution description

Some cleanup and enhancements, namely:

- make path attribute const
- add asserts, to verify path, i.e., must start with `/`

### Issues/PRs references

None

### CODE REVIEW

File: sys/include/net/gcoap.h [hunk 1]
```diff
@@ -504,11 +504,14 @@ void gcoap_register_listener(gcoap_listener_t *listener);
  * @param[in] code      Request code: GCOAP_[GET|POST|PUT|DELETE]
  * @param[in] path      Resource path, *must* start with '/'
  *
+ * @pre `path != NULL`
+ * @pre `path[0] == '/'`
```
Comment (miri64) [inline sys/include/net/gcoap.h]: I would write something like that verbatim, like "Path *must* start with a `/`"

Comment (smlng) [reply sys/include/net/gcoap.h]: agreed, but I often see this notation in our doc, too. Anyway, then the comment can be removed from the param above, right?

File: sys/include/net/gcoap.h [hunk 2]
```diff
@@ -502,13 +502,16 @@ void gcoap_register_listener(gcoap_listener_t *listener);
  * @param[out] buf      Buffer containing the PDU
  * @param[in] len       Length of the buffer
  * @param[in] code      Request code: GCOAP_[GET|POST|PUT|DELETE]
- * @param[in] path      Resource path, *must* start with '/'
```
Comment (miri64) [inline sys/include/net/gcoap.h]: Why did you remove this now. It is always good to have redundant documentation for people with tunnel vision ;-).

**Comment by smlng** [comment]
@miri64 may I squash this?
